core/ErrorQueue.cc

```diff
@@ -176,15 +143,20 @@ bool ErrorQueue::isEmpty() {
     checkOwned();
     return collected.empty();
 }
-vector<unique_ptr<core::ErrorQueueMessage>> ErrorQueue::drainAll() {
+
+UnorderedMap<core::FileRef, vector<unique_ptr<core::ErrorQueueMessage>>> ErrorQueue::drainAll() {
     checkOwned();
-    auto out = drainFlushed();
+    UnorderedMap<core::FileRef, vector<unique_ptr<core::ErrorQueueMessage>>> out;
+
+    core::ErrorQueueMessage msg;
+    for (auto result = queue.try_pop(msg); result.gotItem(); result = queue.try_pop(msg)) {
+        collected[msg.whatFile].emplace_back(make_unique<ErrorQueueMessage>(move(msg)));
+    }
 
     for (auto &part : collected) {
```

Comment:
Can we replace this loop with:

```c++
auto out = move(collected);
collected.clear();
return out;
```

(I forget if `move` on `absl::UnorderedMap` clears the map; I wouldn't be surprised if it didn't.)